Cargo.toml

```diff
@@ -54,8 +55,8 @@ plotters = { version = "0.3.0", optional = true }
 
 [dev-dependencies]
 bridgetree = "0.4"
-criterion = "0.4" #Pinned: 0.5 depends on clap 4 which has MSRV 1.70
-halo2_gadgets = { git = "https://github.com/QED-it/halo2", branch = "zsa1", features = ["test-dependencies"] }
+criterion = "0.4" # 0.5 depends on clap 4 which has MSRV 1.70
+halo2_gadgets = { git = "https://github.com/QED-it/halo2", branch = "orchardzsa-backward-compatability", features = ["test-dependencies"] }
```

Comment:
Done. Fixed the `hex` and `half` dependencies by removing them from "dev-dependencies". Regarding `hash` - in the upstream (zcash/orchard 0.8.0), they replaced hash with `ahash`, which is only present in "dev-dependencies". As for other duplications (including `halo2_gadgets`), they cannot be removed from "dev-dependencies" because they have different features enabled.